public/app/features/dashboard-scene/scene/DashboardScopesFacade.ts

```diff
@@ -12,11 +13,15 @@ export class DashboardScopesFacade extends ScopesFacade {
     super({
       handler: (facade) => {
         if (reloadOnScopesChange && uid) {
-          locationService.reload();
+          this.reloadDashboard();
         } else {
           sceneGraph.getTimeRange(facade).onRefresh();
         }
       },
     });
   }
+
+  private reloadDashboard() {
+    appEvents.publish(new ReloadDashboardEvent());
+  }
```

Comment:
can we leverage a local scene event here instead of a global event (in case there are two dashboards at the same time)

Reply:
I tried to leverage the local scene event but `DashboardScenePageStateManager` is not part of the scene graph. Passing the instance of `DashboardScenePageStateManager` to `transformSaveModelToScene` and to `createDashboardSceneFromDashboardModel` feels wrong as well. Using `getDashboardScenePageStateManager()` in the behaviors directly lead to circular reference between and there are some nasty side effects.